toolkit/sidechain/domain/src/byte_string.rs

```diff
@@ -84,6 +84,20 @@ macro_rules! bounded_str {
     };
 }
 
+impl<T: Get<u32>> Decode for BoundedString<T> {
+	fn decode<I: parity_scale_codec::Input>(
+		input: &mut I,
+	) -> Result<Self, parity_scale_codec::Error> {
+		let bounded_vec: BoundedVec<u8, T> = Decode::decode(input)?;
```

Comment:
Alternatively decode to String and re-use TryFrom.

Reply:
I though about it, but here I have earlier boundary check, so in theory I will not allocate huge String to check later that it is too big